enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProcedureTestBase.java

```diff
@@ -0,0 +1,341 @@
+    }
+
+    protected void testFailAddUserToRole( AuthSubject subject )
+    {
+        testCallFail( subject, "CALL dbms.addUserToRole('Craig', '" + PUBLISHER + "')",
+                QueryExecutionException.class, AuthProcedures.PERMISSION_DENIED );
+    }
+
+    protected void testFailRemoveUserFromRole( AuthSubject subject )
+    {
+        testCallFail( subject, "CALL dbms.removeUserFromRole('Craig', '" + PUBLISHER + "')",
+                QueryExecutionException.class, AuthProcedures.PERMISSION_DENIED );
+    }
+
+    protected void testFailDeleteUser( AuthSubject subject )
+    {
+        testCallFail( subject, "CALL dbms.deleteUser('Craig')", QueryExecutionException.class,
+                AuthProcedures.PERMISSION_DENIED );
+        testCallFail( subject, "CALL dbms.deleteUser('')", QueryExecutionException.class,
+                AuthProcedures.PERMISSION_DENIED );
+    }
+
+    protected void testSuccessfulListUsers( AuthSubject subject, String[] users )
+    {
+        testResult( subject, "CALL dbms.listUsers() YIELD value AS users RETURN users",
+                r -> resultKeyIsArray( r, "users", users ) );
+    }
+
+    protected void testFailListUsers( AuthSubject subject, int count )
+    {
+        testCallFail( subject,
+                "CALL dbms.listUsers() YIELD value AS users RETURN users",
+                QueryExecutionException.class, AuthProcedures.PERMISSION_DENIED );
+    }
+
+    protected void testSuccessfulListRoles( AuthSubject subject, String[] roles )
+    {
+        testResult( subject, "CALL dbms.listRoles() YIELD value AS roles RETURN roles",
+                r -> resultKeyIsArray( r, "roles", roles ) );
+    }
+
+    protected void testFailListRoles( AuthSubject subject )
+    {
+        testCallFail( subject,
+                "CALL dbms.listRoles() YIELD value AS roles RETURN roles",
+                QueryExecutionException.class, AuthProcedures.PERMISSION_DENIED );
+    }
+
+    protected void testFailListUserRoles( AuthSubject subject, String username )
+    {
+        testCallFail( subject,
+                "CALL dbms.listRolesForUser('" + username + "') YIELD value AS roles RETURN count(roles)",
+                QueryExecutionException.class, AuthProcedures.PERMISSION_DENIED );
+    }
+
+    protected void testFailListRoleUsers( AuthSubject subject, String roleName )
+    {
+        testCallFail( subject,
+                "CALL dbms.listUsersForRole('" + roleName + "') YIELD value AS users RETURN count(users)",
+                QueryExecutionException.class, AuthProcedures.PERMISSION_DENIED );
+    }
+
+    protected List<Object> getObjectsAsList( Result r, String key )
+    {
+        return r.stream().map( s -> s.get( key ) ).collect( Collectors.toList() );
+    }
+
+    protected void resultKeyIs( Result r, String key, String... items )
+    {
+        resultKeyIsArray( r, key, items );
+    }
+
+    protected void resultKeyIsArray( Result r, String key, String[] items )
```

Comment:
Since the purpose of this method is to make an assertion, I wonder if naming the method assert* might be useful?